File: src/main/java/com/deswaef/weakauras/mvc/SitemapController.java
package com.deswaef.weakauras.mvc;

import com.deswaef.weakauras.mvc.sitemap.SitemapGenerator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import static org.springframework.web.bind.annotation.RequestMethod.GET;

@Controller
@RequestMapping(value = "/sitemap.xml")
public class SitemapController {

    @Autowired
    private SitemapGenerator sitemapGenerator;

    @RequestMapping(method = GET)
    @ResponseBody
    public String generate() {
        return sitemapGenerator.generate();
    }

}

File: src/main/scala/com/deswaef/wowscrappie/mvc/controller/SitemapController.scala
package com.deswaef.wowscrappie.mvc.controller

import com.deswaef.weakauras.mvc.sitemap.SitemapGenerator
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMethod.GET
import org.springframework.web.bind.annotation.{RequestMapping, ResponseBody}

@Controller
class SitemapController {

  @Autowired private val sitemapGenerator: SitemapGenerator = null

  @RequestMapping(value = Array("/sitemap.xml"), method = Array(GET))
  @ResponseBody def get: String = sitemapGenerator.generate

}
